File: sol/solution/7 Kyu/544aed4c4a30184e960010f4/Solution.hs
{-
Solution
-}
module Divisors where
divisors :: (Show a, Integral a) => a -> Either String [a]
divisors n =
  let ds = [x | x <- [2 .. n - 1], n `mod` x == 0]
  in if null ds
     then Left (show n ++ " is prime")
     else Right ds





{-
Sample Tests
-}
module Codewars.Kata.AlternatingCaseSpec (spec, main) where
module DivisorsSpec where
import Divisors (divisors)
import Test.Hspec

spec :: Spec
spec = do
  describe "divisors" $ do
    it "should work for simple examples" $ do
      divisors 3   `shouldBe` Left "3 is prime"
      divisors 4   `shouldBe` Right [2]
      divisors 13  `shouldBe` Left "13 is prime"
      divisors 15  `shouldBe` Right [3,5]
      divisors 24  `shouldBe` Right [2,3,4,6,8,12]
      divisors 29  `shouldBe` Left "29 is prime"
      divisors 253 `shouldBe` Right [11,23]

File: sol/solution/7 Kyu/544aed4c4a30184e960010f4/Solution.cs
/*
* Solution
*/
public class Kata
{
    public static int[] Divisors(int n)
    {
        var divisorsList = new System.Collections.Generic.List<int>();
        for (int i = 2; i <= n / 2; i++)
        {
            if (n % i == 0)
            {
                divisorsList.Add(i);
            }
        }
        if (divisorsList.Count == 0)
        {
            return null; 
        }
        return divisorsList.ToArray();
    }
}





/*
* Sample Tests
*/
namespace Solution 
{
  using NUnit.Framework;
  using System;

  [TestFixture]
  public class SolutionTest
  {
    [Test]
    public void SampleTest()
    {
      Assert.That(Kata.Divisors(15), Is.EqualTo(new int[] {3, 5}), "n = 15");
      Assert.That(Kata.Divisors(16), Is.EqualTo(new int[] {2, 4, 8}), "n = 16");
      Assert.That(Kata.Divisors(253), Is.EqualTo(new int[] {11, 23}), "n = 253");
      Assert.That(Kata.Divisors(24), Is.EqualTo(new int[] {2, 3, 4, 6, 8, 12}), "n = 24");
      Assert.That(Kata.Divisors(7), Is.Null, "n = 7");
    }
  }
}